share/shutter/resources/modules/Shutter/Pixbuf/Save.pm

```diff
@@ -116,56 +131,29 @@ sub save_pixbuf_to_file {
 		};
 	} elsif ($filetype eq 'png') {
 
-		#get quality value from settings if not set
-		if (my $settings = $self->{_common}->get_globalsettings_object) {
-			if (defined $settings->get_image_quality("png")) {
-				$quality = $settings->get_image_quality("png");
-			} else {
-				$quality = 9;
-			}
-		} else {
-			$quality = 9;
-		}
+		set_quality_setting($filetype);
```

Comment:
You're passing a single parameter, so `$filetype` becomes `$self`.

Call this function as a method instead

Reply:
Right, thanks!